### ISSUE TITLE
Deprecate and Remove audit.k8s.io/v1[alpha|beta]1 Versions

### ISSUE THREAD

**Issue by vpnachev** [kubernetes/kubernetes#98035]
**What would you like to be added**:
Since k8s v1.12 the API audit.k8s.io has a stable version V1, ref: https://github.com/kubernetes/kubernetes/pull/65891.
However, the v1alpha1 and v1beta1 versions of the API are still not deprecated and available
- https://github.com/kubernetes/kubernetes/tree/v1.21.0-alpha.1/staging/src/k8s.io/apiserver/pkg/apis/audit/v1alpha1
- https://github.com/kubernetes/kubernetes/tree/v1.21.0-alpha.1/staging/src/k8s.io/apiserver/pkg/apis/audit/v1beta1

If they got removed, the kube-apiserver flags `--audit-log-version` and `--audit-webhook-version` will accept only single valid value `audit.k8s.io/v1`, so these flags can be also deprecated at the moment, however I lean toward keeping them in case a new API version is implemented in the future.

/area audit
/sig auth
/kind deprecation
/kind cleanup
/kind api-change

**Comment by liggitt** [comment]
cc @soltysh @tallclair 

Marking the alpha/beta version as deprecated and starting the clock on them seems reasonable. That would also mean warning if a version other than `audit.k8s.io/v1` was passed to those two flags.

**Comment by carlory** [comment]
@liggitt  I send a pr about it, but it failed in some tests. Can you give me some guide?  how to deprecate some audit versions located at  `staging/src/k8s.io/apiserver`?

**Comment by vpnachev** [comment]
Now when https://github.com/kubernetes/kubernetes/pull/98858 is merged, should this issue be closed as fixed?

**Comment by carlory** [comment]
@vpnachev  audit.k8s.io/v1[alpha|beta]1 versions will be removed in v1.24. 

release notes:

```
`audit.k8s.io/v1beta1` and `audit.k8s.io/v1alpha1` audit policy configuration and audit events are deprecated in favor of `audit.k8s.io/v1`, available since v1.13. kube-apiserver invocations that specify alpha or beta policy configurations with `--audit-policy-file`, or explicitly request alpha or beta audit events with `--audit-log-version` / `--audit-webhook-version` must update to use `audit.k8s.io/v1` and accept `audit.k8s.io/v1` events prior to v1.24.
```

**Comment by vpnachev** [comment]
> Now when #98858 is merged, should this issue be closed as fixed?

Still valid question. @liggitt any opinion? Should it stay open till v1[alpha|beta]1 are removed in 1.24 or we can close it now?

/remove-lifecycle stale

**Comment by ialidzhikov** [comment]
> Still valid question. @liggitt any opinion? Should it stay open till v1[alpha|beta]1 are removed in 1.24 or we can close it now?

ping @liggitt 

/remove-lifecycle stale

**Comment by liggitt** [comment]
we can keep it open until then

**Comment by doshyt** [comment]
@liggitt - I'm from Bug triage for 1.24. This issue caught my eye since there were no updates for a long time. It seems everything is done and ready to be wrapped. Do you prefer this issue to be closed after 1.24 is released or now while we are in it? Thanks!
